### PR TITLE
Improve testing

### PR DESCRIPTION
**Pull request by AsaAyers**
This PR is working on solving #19.

First, I moved to Jest as that's what we're using for other projects.

The goal of `test-example` is to verify the exposed API of fetch-tree. Ideally, it will have an example of every feature fetch-tree provides and this can be tracked with the coverage report. I think it would be really nice if it also serves as additional documentation and a reference for how you might structure a project that uses fetch-tree.

This branch will get squashed/rebased before it gets merged.

### CODE REVIEW

**Comment by AsaAyers** [comment]
I'm considering this done:

selection_030

File: test-example/tests/example-mistakes.test.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    test(`loader action must return a promise`, async () => {
+        const Example = fetchTree({
+            component: Dummy,
+            resourceGroup: group({
+                id: fromProps('id'),
+                data: depends(
+                    ['id'],
+                    loader({
+                        id(id) { return `data-${id}` },
+                        // oops, this calculates and ID and then doesn't return it
```
Comment (cbranch101) [inline test-example/tests/example-mistakes.test.js]: This comment needs to be updated to reflect the correct error

File: test-example/tests/example-mistakes.test.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            }),
+        })
+
+        const error = await captureError(
+            <Example id={1} />
+        )
+        expect(error.message).toMatch(/Actions need to return a promise/)
+    })
+
+    test(`depends() Referring to a resource out of order`, async () => {
```
Comment (cbranch101) [inline test-example/tests/example-mistakes.test.js]: These tests are super helpful

File: src/node_types/__tests__/from-props.js [hunk 3]
```diff
@@ -1,56 +0,0 @@
-import test from 'ava'
```
Comment (cbranch101) [inline src/node_types/__tests__/from-props.js]: While not expressly needed to get full coverage, I think these tests were really helpful for documenting exactly what each node did, and might be useful down the road if you decide the change the specific implementation of a given node.  I'm not convinced you need to bring them back now, but I was a little bumped that aside from confirming that errors on thrown on invalid arguments, the example tests are the only way that the specific nodes are being tested.

**Pull request merged**